ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java

```diff
@@ -846,7 +846,16 @@ else if(vote == XAResourceOrphanFilter.Vote.ROLLBACK)
         }
         catch (XAException e1)
         {
-        	jtaLogger.i18NLogger.warn_recovery_xarecovery1(_logName+".xaRecovery", XAHelper.printXAErrorCode(e1), e1);
+            if(e1.errorCode == XAException.XAER_NOTA)
+            {
+                if(jtaLogger.logger.isDebugEnabled()) {
+                    jtaLogger.logger.debug("XAER_NOTA received while rolling back " + XAHelper.xidToString(xid));
```

Comment:
I'm not fully sure here... Would not be good to retain the exception stacktrace in the output even it's kind of expected exception?

Reply:
I'm seeing now _logName+".xaRecovery" is missing, without that it's not clear where the log is coming from, since jtaLogger is shared for all jta classes. For stack trace, that wouldn't hurt either, but would it be consistent with the rest of the codebase, is there any other place where debug log has stacktrace?

Reply:
to me returning use of `_logName` sounds as a good approach for not changing the previous behaviour abruptly

my belief is to use the right and enough information in the log to provide a good way to investigate in case of some failures. I personally do not bound this to the reasoning that there is no such necessity in the code around. For sure there are some (but not often) cases of debugging showing stacktrace in the Narayana codebase (https://github.com/jbosstm/narayana/blob/master/ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/recovery/recoverycoordinators/ResourceCompletor.java#L79)

my doubt was what value it will provide for somebody investigation. I tend to think, now, that's for NOTA it's not necessary as Xid will link the information with prior operations.